pytorch_lightning/trainer/data_loading_mixin.py

```diff
@@ -24,7 +32,7 @@ def init_train_dataloader(self, model):
         self.get_train_dataloader = model.train_dataloader
 
         # determine number of training batches
-        if isinstance(self.get_train_dataloader(), IterableDataset):
+        if EXIST_ITER_DATASET and isinstance(self.get_train_dataloader(), IterableDataset):
```

Comment:
```suggestion
        if EXIST_ITER_DATASET and isinstance(self.get_train_dataloader().dataset, IterableDataset):
```
I think that we want to check the dataset attribute rather than the DataLoader itself (since it contains the IterableDataset). If we fix it here I can remove #547 .